test/e2e/test/kibana/steps_mutation.go

```diff
@@ -5,20 +5,73 @@
 package kibana
 
 import (
+	"fmt"
 	"testing"
 
 	"github.com/elastic/cloud-on-k8s/pkg/apis/kibana/v1beta1"
+	"github.com/elastic/cloud-on-k8s/pkg/controller/common/hash"
 	"github.com/elastic/cloud-on-k8s/pkg/utils/k8s"
 	"github.com/elastic/cloud-on-k8s/test/e2e/test"
 	"github.com/stretchr/testify/require"
+	corev1 "k8s.io/api/core/v1"
 )
 
 func (b Builder) MutationTestSteps(k *test.K8sClient) test.StepList {
-	return b.UpgradeTestSteps(k).
+	return b.AnnotatePodsWithBuilderHash(k).
+		WithSteps(b.UpgradeTestSteps(k)).
 		WithSteps(b.CheckK8sTestSteps(k)).
 		WithSteps(b.CheckStackTestSteps(k))
 }
 
+func (b Builder) AnnotatePodsWithBuilderHash(k *test.K8sClient) test.StepList {
```

Comment:
IIUC this is almost a duplicate from `AnnotatePodsWithBuilderHash` in the elasticsearch package. Except the way we list pods is different. Can we try to factor things together?

Reply:
Also I think it's fine to use the same `BuilderHashAnnotation` for both.

Reply:
It differs in listing pods, annotation name and hash generation which seems like the entire step. I can refactor if you feel strongly, but I didn't hence this way :)

Reply:
Chatted offline, will fix 👍